### PR TITLE
Add Header Card widget

### LINKED ISSUE #978: Improve Card implementation
## Enhancement 

Current card implementation requires the user to use the card css classes from the default theme to apply styling / positioning to a card's child nodes. 
This is not an ideal solution as it does not match patterns used elsewhere in the dojo widgets nor does it allow theming to work appropriately.

We should come up with a better way to do this.
Possible solutions are as follows:

### render props / child render props

This would provide the users with props they can pass to render their content in the correct position.
ie.
```tsx
<Card primaryContent={renderPrimary} secondaryContent={renderSecondary} />
```
The down side of this is that is gives the user no control over where they place each of these render results. 

### Child widgets

This would provide the users with an assortment of smaller widgets they can use to render styled content. ie.
```tsx
<Card>
   <CardPrimary>primary stuff</CardPrimary>
   <CardButtons>{...buttons}</CardButtons>
</Card>
```
This would give users greater flexibility over where they place their children but removes the ability for our card widget to be opinionated about its children's layout / positioning which may be important for some layouts.

### Card derivatives with render / child render props

This, I feel, gives us the best all-around flexibility and control as we can specify different card types, ie. `MediaCard` / `PrimaryCard` / `SubtitleCard` and with such cards, provide the users with a specific, typed childen render object which they can use.
This maintains the ability to be opinionated about layout and removes the need to use theme classes from card. It also encourages users to create their own card templates etc which they would use within their own apps.

```tsx
<MediaCard imgSrc="imgSrc">
{{ 
   title: () => "I am the title",
   buttons: () => <RaisedButton>hello</RaisedButton>
 }}
</MediaCard>
```

```tsx
<SubtitleCard>
{{
   title: () => "card title",
   subtitle: () => "card sub title"
}}}
</SubtitleCard>
```

### PR DESCRIPTION
**Pull request by nicknisi**
**Type:** feature

**Note:** This PR relies on #1105 and #1104 being merged. Please review those first.

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] For new widgets, an entry has been added to the `.dojorc`
* [ ] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**

This adds a new card, a header card, that wraps the card implementation and takes over rendering of the card's `header` named child to provide a consistently-styled header for the card. The header-card takes all the same properties and children (minus `header` named child). It takes over rendering the `title` and `subtitle` if provided and will render them in the header (and prevent them from being passed to the underlying card). It also adds an `avatar` property which will render an avatar in the header next to the title/subtitle.

Resolves #978 

**Example**
```tsx
<HeaderCard
	avatar={() => <Avatar src={avatar} />}
	title="Hello, World"
	subtitle="Lorem ipsum"
	mediaSrc={mediaSrc}
>
	{{
		content: () => <p>Lorem ipsum</p>,
		actionButtons: () => <Button>Action</Button>,
		actionIcons: () => (
			<virtual>
				<Icon type="upIcon" />
				<Icon type="downIcon" />
			</virtual>
		)
	}}
</HeaderCard>
```
image

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-cvwrgwsaw.now.sh
* Tests Deployment: https://dojowidgets-qt63tjh27.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-n58k0wgdu.now.sh
* Tests Deployment: https://dojowidgets-1p5cmyzwm.now.sh

File: src/header-card/index.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	.children<HeaderCardChildren | undefined>();
+
+export const HeaderCard = factory(function HeaderCard({
+	middleware: { theme },
+	properties,
+	children
+}) {
+	const themeCss = theme.classes(css);
+	const { title, subtitle, avatar, ...cardProps } = properties();
+	const cardChildren = children()[0];
```
Comment (agubler) [inline src/header-card/index.tsx]: Could this be `const [ cardChildren ] = children();`

Comment (tomdye) [inline src/header-card/index.tsx]:
```suggestion
	const [ cardChildren ] = children();
```

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-qef3ylg2b.now.sh
* Tests Deployment: https://dojowidgets-1epjlfke5.now.sh

**Pull request merged**